src/runtime/hexagon_remote/elf.h

```diff
@@ -143,7 +159,8 @@ struct elf_t {
         // stuff, the same size again to make a writeable copy, and
         // then an extra page for the global offset table for PIC
         // code.
-        buf = (char *)mmap(NULL, size * 2 + global_offset_table_size,
+        mmap_ptr = (mmap_ptr_fn) halide_get_symbol("mmap");
```

Comment:
i tried that approach, but the library as built has a GLOBAL FUNC dependency against mmap. mprotect, and munmap. Even though i added weak, the version found in libc  (-lc) would override the weak. Thus the library would fail to load.